tests/test_model.py

```diff
@@ -522,3 +522,417 @@ def test_model_diskonlyfilemodel_field_is_constructed_correctly(
+            assert any(
+                line.strip() == line_to_check for line in file
+            ), f"File {save_location} does not contain the line: {line_to_check}"
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(None),
+            pytest.param(1),
+        ],
+    )
+    def test_dimr_with_fmcomponent_saving_process_when_process_should_be_left_out(
+        self, tmp_path, input_process
+    ):
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            process=input_process,
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+
+        dimr = DIMR(component=component)
+        save_location: Path = tmp_path / "dimr_config.xml"
+        dimr.save(filepath=save_location)
+
+        process = "<process>"
+
+        with open(save_location, "r") as file:
+            assert (
+                process not in file
+            ), f"File {save_location} does contain the line: {process}"
+
+
+class TestFmComponentProcess:
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(None),
+            pytest.param(1),
+            pytest.param(2),
+            pytest.param(3),
+            pytest.param(4),
+            pytest.param(5),
+        ],
+    )
+    def test_fmcomponent_process_after_init(self, input_process: int):
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            process=input_process,
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        assert component.process == input_process
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(None),
+            pytest.param(1),
+            pytest.param(2),
+            pytest.param(3),
+            pytest.param(4),
+            pytest.param(5),
+        ],
+    )
+    def test_fmcomponent_process_after_setting(self, input_process: int):
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        component.process = input_process
+        assert component.process == input_process
+
+    def test_fmcomponent_without_process_after_init(self):
+        expected_process_format = None
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        assert component.process == expected_process_format
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param("lalala"),
+            pytest.param("123"),
+            pytest.param(3.5),
+        ],
+    )
+    def test_fmcomponent_process_after_init_with_incorrect_input_throws_valueerror(
```

Comment:
-3 and 0 are also incorrect values. Maybe use `invalid type` somwhere in the test instead?

Reply:
will update